core/src/main/kotlin/builder/kord/KordProxyBuilder.kt

```diff
@@ -0,0 +1,44 @@
+package dev.kord.core.builder.kord
+
+import dev.kord.cache.api.DataCache
+import dev.kord.common.annotation.KordExperimental
+import dev.kord.common.entity.Snowflake
+import dev.kord.core.ClientResources
+import dev.kord.core.Kord
+import dev.kord.core.gateway.DefaultMasterGateway
+import dev.kord.core.supplier.EntitySupplyStrategy
+import dev.kord.gateway.Gateway
+import dev.kord.gateway.builder.Shards
+import dev.kord.rest.service.RestClient
+import kotlinx.coroutines.flow.MutableSharedFlow
+
+/**
+ * The rest only Kord builder. You probably want to invoke the [DSL builder][Kord.restOnly] instead.
+ */
+@KordExperimental
+public class KordProxyBuilder: RestOnlyBuilder() {
+
+    public override fun build(): Kord {
+        val client = httpClient.configure()
+
+        val resources = ClientResources(
+            "",
+            applicationId ?: Snowflake.min,
```

Comment:
I think it's problematic to provide `Snowflake.min` as a default, we use the application id in some places. Should we throw an exception instead?

Reply:
I can't think of a better alternative tbf; the id is required so

Reply:
we can't cause someone may not supply any

Reply:
but then the bot may not function correctly and throw in other places

Reply:
should we create a mocked version of some sorts?
@WinteryFox would you supply an applicationId here?
I mean as a Proxy Kord Builder we still need some applicationId to work on so @Lukellmann it does make sense to throw

Reply:
I do supply an application ID, you can just require it in the constructor. Since I'm currently still using the workaround of multiple Kord instances its fine for now, but should probably get a more long-term fix in the future.